### PR TITLE
Fix: TCR Adapter namespcae check.

### PR DESCRIPTION
**Pull request by kofj**
Fix & UT

1. Fix TCR Adapter namespcae check.
2. Add Chart UT.

Signed-off-by: 疯魔慕薇 <kfanjian@gmail.com>

### CODE REVIEW

**Comment by chlins** [comment]
@kofj Is this pr for fixing https://github.com/goharbor/harbor/issues/16050?

File: src/pkg/reg/adapter/tencentcr/sdk.go [hunk 1]
```diff
@@ -131,10 +131,12 @@ func (a *adapter) isNamespaceExist(namespace string) (exist bool, err error) {
 	}
 
 	log.Warningf("[tencent-tcr.PrepareForPush.isNamespaceExist] namespace=%s, total=%d", namespace, *resp.Response.TotalCount)
-	if int(*resp.Response.TotalCount) != 1 {
-		return
+	for _, ns := range resp.Response.NamespaceList {
+		if ns.Name == &namespace {
```
Comment (chlins) [inline src/pkg/reg/adapter/tencentcr/sdk.go:135]: Are two pointers compared as expected?

Comment (kofj) [reply src/pkg/reg/adapter/tencentcr/sdk.go]: Fixed

**Comment by kofj** [comment]
> 

Yean, I'll back-port latest 3 version.

**Comment by chlins** [comment]
@kofj Thanks, pls rebase the latest main to resolve the CodeQL alerts in CI.

**Review by chlins — APPROVED** [review]
lgtm

**Review by chlins — APPROVED** [review]
lgtm

**Comment by kofj** [comment]
@wy65701436 @chlins review again pls.

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**